cirq-core/cirq/circuits/circuit.py

```diff
@@ -914,6 +914,9 @@ def all_measurement_key_names(self) -> AbstractSet[str]:
     def _measurement_key_names_(self) -> AbstractSet[str]:
         return self.all_measurement_key_names()
 
+    def _measurement_key_objs_(self) -> AbstractSet[value.MeasurementKey]:
```

Comment:
May be useful to override in frozencircuit with a caching version.

Reply:
Done. Only caching the objects now since the transformation from object to string is pretty lightweight and doesn't seem worth it to bloat the object size for it.